lib/modules/apostrophe-utils/lib/api.js

```diff
@@ -622,4 +623,52 @@ module.exports = function(self, options) {
     self.logger.error.apply(self.logger, arguments);
   };
 
+  // Performance profiling method. At the start of the operation you want
+  // to profile, call with req (may be null or omitted entirely) and a
+  // dot-namespaced key. A function is returned. Call that function
+  // with no arguments at the end of your operation.
+  //
+  // Alternatively, you may pass the duration in milliseconds yourself as the
+  // third argument. In this case no function is returned. This is useful if
+  // you are already gathering timing information for other purposes.
+  //
+  // Profiler modules such as `apostrophe-profiler` override this
+  // method to provide detailed performance analysis. Note that they must
+  // support both calling syntaxes. The default implementation does nothing.
+  //
+  // If the dot-separated key looks like `callAll.pageBeforeSend.module-name`,
+  // time is tracked to `callAll`, `callAll.pageBeforeSend`, and
+  // `callAll.pageBeforeSend.module-name` as categories. Note that the
+  // most general category should come first.
+  //
+  // To avoid overhead and bloat in the core, the default implementation
+  // does nothing. Also most core modules and methods do not invoke this method.
+  // However, the `apostrophe-profiler` module extends them to invoke it,
+  // for performance reasons: the profiler itself can have
+  // a performance overhead.
+
+  self.profile = function(req, key /* , duration */) {
+    if ((typeof req) === 'string') {
+      if (arguments.length === 2) {
+        return self.profile(null, arguments[0], arguments[1]);
+      } else {
+        return self.profile(null, arguments[0]);
+      }
+    }
+    if (arguments.length === 2) {
+      // This is a do-nothing placeholder implementation
+      return function() {};
+    } else {
+      // All-in-one version with duration, there's nothing to fake
+    }
+  };
+
+  // Returns time since the start of the process in milliseconds,
+  // with high-resolution accuracy. Used by apos.utils.profile.
```

Comment:
`Used by apos.utils.profile.` is confusing since it doesn't seem to actually be used.